File: frontend/src/services/auth.service.js
import api from './api'

export const authService = {
  login: async (username, password) => {
    const response = await api.post('/auth/login', { username, password })
    return response.data
  },
  
  register: async (userData) => {
    const response = await api.post('/auth/register', userData)
    return response.data
  },
  
  logout: async () => {
    const response = await api.post('/auth/logout')
    return response.data
  },
  
  getProfile: async () => {
    const response = await api.get('/auth/profile')
    return response.data
  },
}

File: backend/src/main/java/com/bharaterp/service/AuthService.java
package com.bharaterp.service;

import org.springframework.stereotype.Service;

@Service
public class AuthService {
    public String login(String username, String password) {
        // Authentication logic
        return "Login successful";
    }
    
    public String register(String username, String password, String email) {
        // Registration logic
        return "Registration successful";
    }
    
    public String logout(String token) {
        // Logout logic
        return "Logout successful";
    }
}
